teachertool/src/components/MakecodeFrame.tsx

```diff
@@ -1,140 +1,31 @@
 /// <reference path="../../../built/pxteditor.d.ts" />
 import * as React from "react";
-import { useState, useEffect, useRef } from "react";
-import { ProgressBar } from "react-common/components/controls/ProgressBar";
+import { useEffect, useRef } from "react";
+import { getIframeRef, removeIframeRef } from "../services/makecodeEditorService";
 interface MakeCodeFrameProps {
     pageSourceUrl: string;
-    highContrast?: boolean;
-    onFrameOpen: () => void;
-    onFrameClose: () => void;
     tutorialEventHandler?: (event: pxt.editor.EditorMessageTutorialEventRequest) => void;
 }
 
-type FrameState = "loading" | "no-project" | "opening-project" | "project-open" | "closing-project";
-
-interface PendingMessage {
-    original: pxt.editor.EditorMessageRequest;
-    handler: (response: any) => void;
-}
-
 export const MakeCodeFrame: React.FC<MakeCodeFrameProps> =
-    ( { pageSourceUrl,
-        highContrast,
-        onFrameClose,
-        onFrameOpen
-    } ) => {
-    let ref: HTMLIFrameElement | undefined;
-    const messageQueue: pxt.editor.EditorMessageRequest[] = [];
-    let nextId: number = 0;
-    let pendingMessages: {[index: string]: PendingMessage} = {};
-    const [frameState, setFrameState] = useState<FrameState>("loading");
-    const [loadPercent, setLoadPercent] = useState(0);
-    const [workspaceReady, setWorkspaceReady] = useState(false);
+    ( { pageSourceUrl} ) => {
 
     let frameRef = useRef<HTMLIFrameElement>(null);
 
     useEffect(() => {
-        // logic we want to do when the iframe is loaded
-        if (ref && ref.contentWindow) {
-            window.addEventListener("message", onMessageReceived);
-            ref.addEventListener("load", handleFrameReload)
-
+        if (frameRef.current) {
+            getIframeRef(frameRef.current);
         }
 
-        // logic we want when the iframe unmounts
         return () => {
-            window.removeEventListener("message", onMessageReceived);
-            if (ref && ref.contentWindow) {
-                ref.removeEventListener("load", handleFrameReload)
+            if (frameRef.current) {
```

Comment:
This check is unnecessary. It should be safe to always call `removeIFrameRef` here.